### PR TITLE
Allow admins/supervisors to generate court reports for a case

### LINKED ISSUE #1969: let supervisors and admins generate court reports for a case
**What type of user is this for? volunteer/supervisor/admin/all**
volunteer, supervisor, admin

**Description**
https://casa-qa.herokuapp.com/case_court_reports
Let supervisors and admins generate court reports for a case
Add generate court report button to a case as well as having it in a sidebar item

**Screenshots of current behavior, if any**
Screen Shot 2021-04-21 at 6 35 06 PM

_QA Login Details:_
url: https://casa-qa.herokuapp.com/
password for all users: 123456

Log in using any of the below emails, depending on the account type you'd like to log in as.
- volunteer1@example.com
- supervisor1@example.com
- casa_admin1@example.com

### PR DESCRIPTION
**Pull request by librod89**
### What github issue is this PR for, if any?
Resolves #1969

### What changed, and why?
- Added the "Generate Court Reports" link in the sidebar for admin and supervisors
- Added a "Generate Report" button to the casa_cases#show page

### How will this affect user permissions?
Supervisors and Admin can now access the Generate Court Reports page and can generate reports for CasaCases within their organization

### How is this tested? (please write tests!) 💖💪

- Request tests for case_court_reports controller
- System tests for casa_cases#show page

### Screenshots please :)

image

### CODE REVIEW

File: app/controllers/case_court_reports_controller.rb [hunk 1]
```diff
@@ -59,12 +58,20 @@ def set_casa_case
     @casa_case = CasaCase.find_by(case_number: params[:id])
   end
 
+  def assigned_cases
+    @assigned_cases = if current_user.volunteer?
+                        CasaCase.actively_assigned_to(current_user)
+                      else
+                        current_user.casa_org.casa_cases.active
```
Comment (librod89) [inline app/controllers/case_court_reports_controller.rb:65]: I assumed here that we only wanted admin/supervisors to generate reports for active cases in their organization. Is that right?

File: app/controllers/case_court_reports_controller.rb [hunk 2]
```diff
@@ -59,12 +58,20 @@ def set_casa_case
...
+                      end
+  end
+
   def generate_report_to_string(casa_case)
     return unless casa_case
 
     casa_case.casa_org.open_org_court_report_template do |template_docx_file|
       court_report = CaseCourtReport.new(
-        volunteer_id: current_user.id, # ??? not a volunteer ? linda
+        volunteer_id: current_user.volunteer? ? current_user.id : casa_case.assigned_volunteers.first.id,
```
Comment (librod89) [inline app/controllers/case_court_reports_controller.rb:74]: A volunteer is required to generate a report. So I'm always picking the first assigned volunteer. Is there another way you'd like to solve this problem?

File: app/javascript/src/casa_case.js [hunk 3]
```diff
@@ -84,7 +84,68 @@ function courtMandateHtml (index) {
   }
 }
 
+function showBtn(el) { el.classList.remove('d-none') }
```
Comment (librod89) [inline app/javascript/src/casa_case.js:87]: All of this javascript is directly moved from `app/views/case_court_reports/index.html.erb` for the purposes of reusing it. The only change is how I set the `url` on line 114

File: app/javascript/src/casa_case.js [hunk 4]
```diff
@@ -84,7 +87,67 @@ function courtMandateHtml (index) {
   }
 }
 
+function showBtn (el) { el.classList.remove('d-none') }
```
Comment (librod89) [inline app/javascript/src/casa_case.js:90]: All of this javascript is directly moved from app/views/case_court_reports/index.html.erb for the purposes of reusing it. The only change is how I set the url on line 114

File: app/controllers/case_court_reports_controller.rb [hunk 5]
```diff
@@ -59,12 +58,20 @@ def set_casa_case
     @casa_case = CasaCase.find_by(case_number: params[:id])
   end
 
+  def assigned_cases
+    @assigned_cases = if current_user.volunteer?
+      CasaCase.actively_assigned_to(current_user)
+    else
+      current_user.casa_org.casa_cases.active
```
Comment (librod89) [inline app/controllers/case_court_reports_controller.rb:65]: I assumed here that we only wanted admin/supervisors to generate reports for active cases in their organization. Is that right?

File: cypress/integration/volunteer/generate_court_report.spec.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ context("Logging into cypress as a volunteer", () => {
   it("should generate a court report", () => {
     cy.get("#toggle-sidebar-js").click();
     cy.contains("Generate Court Reports").click();
-    cy.get("#case-selection").select("CINA-18-1003 - non-transition");
+    cy.get("#case-selection").select("CINA-18-1003 - transition");
```
Comment (librod89) [inline cypress/integration/volunteer/generate_court_report.spec.js:11]: This change is unrelated. The seed data just happened to reach age 16 yesterday 😄

File: cypress/integration/volunteer/generate_court_report.spec.js [hunk 7]
```diff
@@ -8,10 +8,16 @@ context("Logging into cypress as a volunteer", () => {
   it("should generate a court report", () => {
     cy.get("#toggle-sidebar-js").click();
     cy.contains("Generate Court Reports").click();
-    cy.get("#case-selection").select("CINA-18-1003 - non-transition");
```
Comment (librod89) [inline cypress/integration/volunteer/generate_court_report.spec.js:11]: The changes in this test are not related. But I didn't realize that this test that I wrote yesterday was failing on master. I have fixed it by removing the static string so it can select whatever case is in the dropdown.

File: app/controllers/case_court_reports_controller.rb (hunk 5 above)
Comment (compwron) [reply app/controllers/case_court_reports_controller.rb:65]: I'm not sure. In general we want to be careful with leaving data out- it is possible that a case was active but only recently became inactive, and should still be included in a report.

File: app/controllers/case_court_reports_controller.rb (hunk 2 above)
Comment (compwron) [reply app/controllers/case_court_reports_controller.rb:74]: This seems reasonable for now because most cases have only one volunteer. Eventually we will want to be able to specify the volunteer, since some cases have multiple volunteers (think sibling youth, and a married pair of volunteers)- and I'm not sure whether both volunteers need to submit a court report for the same youth.

**Review by compwron — APPROVED** [review]
Very nice! :)
